pennylane/_qubit_device.py

```diff
@@ -1314,9 +1317,10 @@ def expval(self, observable, shot_range=None, bin_size=None):
         if self.shots is None:
             try:
                 eigvals = self._asarray(
-                    observable.eigvals()
-                    if not isinstance(observable, MeasurementValue)
-                    else np.arange(2 ** len(observable.wires)),
+                    observable.eigvals() if not isinstance(observable, MeasurementValue)
+                    # Indexing a MeasurementValue gives the output of the processing function
+                    # for that index as a binary number.
+                    else [observable[i] for i in range(2 ** len(observable.wires))],
```

Comment:
It works because `defer_measurements` maps the wires to the new ones. This could potentially be problematic in the future though, and it's a simple enough change. Thanks Matt!